pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -121,6 +121,12 @@ func (n namedRangerEntry) Network() net.IPNet {
 
 // onNetworkChange is fired if the default network is changed either via the namespace label or mesh-networks
 func (c *Controller) onNetworkChange() {
+	// instead of pushing for each pod/endpoint, trigger push at last
```

Comment:
Why does it matter? debounce will handle it anyways?

I would strongly prefer that we handle this in the controller instead of a "pause" mechanism (if debounce isn't sufficient). Like make the `sync` functions have a flag to skip push would give the same behavior without the complex pause+resume.

Reply:
But also this is a regression on the precision of the ConfigUpdates. This will ALWAYS push a full NetworksTrigger push, but that only happens previously if `gwsChanged`.

So now all users will get full push even if they don't use multi-networks or the GW is unchanged

Reply:
`onNetworkChange` is a cb of meshNetworks Watcher and system namespace handler, a full NetworksTrigger is not a big deal.

>I would strongly prefer that we handle this in the controller instead of a "pause" mechanism (if debounce isn't sufficient). Like make the sync functions have a flag to skip push would give the same behavior without the complex pause+resume.

How to add a flag to sync function? We must add for many functions in the calling chain

There are some cons:

1. the metrics recorded is not right
2. debounce mechanism can reduce push times, but in case of large scale mesh, there could be so many pods/endpoints, not sure how long will this take and how many times we need to push

Reply:
Debounce will max at 10s. If this takes more than 10s it may actually be unsafe - we may entirely block pushes globally for a long time while we process everything.

So it's either fast enough debounce should work or slow enough that it's unsafe to pause the world IMO

Reply:
I would argue another sideeffect if we push multi times during the network change.

This could generate incorrect xds, typically some endpoints are with network{old}, some with network{new}